### ISSUE TITLE
A11y : Switch component does not have a focus state

### ISSUE THREAD

**Issue by yaelleC** [grafana/grafana#46472]
**Steps to reproduce**:
When tab-ing our way through the datasource creation page, we will lose focus between a textbox and another (where the switch is). 
Non blocking as we can still toggle the switch but very tedious in `Auth` section as it's 6 switch and we don't see where we are when using tabs

**Actual Result**:
switch not focussed

**Expected Result**
we should see the switch focus state (highlighted) . This could be achieved using the `InlineSwitch` component instead of `Switch` or by modifying `Switch` to be accessible.
Gio Ricci started a PR for that during the Hackathon week: #46252 

**Relevant WCAG Criteria:** 
[2.4.7 Focus visible - Level AA](https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0&showtechniques=335%2C332#focus-visible)

**Environment**:
- Grafana version: 8.4.3
